### PR TITLE
AG-7622 - Add version mismatch checks

### PR DESCRIPTION
**Pull request by taktran**
## Changes

* Add errors in the console when
  * Grid modules/packages are using different versions
  * Grid module and Standalone Charts version are incompatible
* Update `Module` interface to have `version` numbers and allow for custom validations (`validate` function)
* Update `versionModules` script to update `version.ts` files based on version number, to be run during build time. The `version.ts` files are added manually initially. This allows `chartModelMigration` version update code/scripts to be removed as it is generalised to all modules/packages.

## Error messages

Eg,

Screenshot 2022-12-09 at 5 01 20 pm

- Incompatible charts version
  
    `AG Grid: AG Grid version 28.2.1 and AG Charts version 10.2.1 is not supported. AG Grid version 28.2.x should be used with AG Chart 6.2.x. Please see https://www.ag-grid.com/javascript-data-grid/modules/ for more information.`
    
- `GridChartsModule` - charts version is `undefined`
    
    `AG Grid: AG Grid version 28.2.x should be used with AG Chart 6.2.x. Please see https://www.ag-grid.com/javascript-data-grid/modules/ for more information.`
    
- `GridChartsModule` - grid version is `undefined` (shouldn’t really ever happen)
    
    `AG Grid: AG Grid version is incompatible. Please see https://www.ag-grid.com/javascript-data-grid/modules/ for more information.`
    
- Incompatible grid module
    
    `AG Grid: You are using incompatible versions of AG Grid modules. Major and minor versions should always match across modules. '@ag-grid-enterprise/range-selection' is version 30.2.1 but the other modules are version 28.2.1. Please update all modules to the same version.`
    
- Grid module has `undefined` version
    
    `AG Grid: You are using incompatible versions of AG Grid modules. Major and minor versions should always match across modules. '@ag-grid-enterprise/range-selection' is incompatible. Please update all modules to the same version.`

### CODE REVIEW

File: community-modules/core/src/ts/modules/moduleRegistry.ts [hunk 1]
```diff
@@ -8,8 +8,26 @@ export class ModuleRegistry {
...
+            console.error(`AG Grid: You are using incompatible versions of AG Grid modules. Major and minor versions should always match across modules. '${module.moduleName}' is version ${module.version} but the other modules are version ${this.currentModuleVersion}. Please update all modules to the same version.`);
+        }
+
+        if (module.validate) {
+            const result = module.validate();
+            if (!result.isValid) {
+                console.error(`AG Grid: ${result.message}`);
+            }
+        }
+
```
Comment (seanlandsman) [inline community-modules/core/src/ts/modules/moduleRegistry.ts:30]: this new block could probably do with being extracted into a new validation method

File: grid-packages/ag-grid-docs/documentation/doc-pages/grid-api/doc-interfaces.AUTO.json [hunk 2]
```diff
@@ -14213,7 +14213,17 @@
       "type": { "arguments": {}, "returnType": "void", "optional": false }
     }
   },
+  "ModuleValidationResult": {},
   "Module": {
+    "version": { "type": { "returnType": "string", "optional": false } },
+    "validate": {
```
Comment (seanlandsman) [inline grid-packages/ag-grid-docs/documentation/doc-pages/grid-api/doc-interfaces.AUTO.json:14219]: I don't think we should publicise this - it's not really meant for public consumption.

This is probably out of scope for this change but highlighting where & how we validation the license isn't a good idea imo

**Review by seanlandsman — COMMENTED** [review]
I think it'll probably be a good idea of Alan was added to review the chart related changes

**Review by alantreadway — COMMENTED** [review]
LGTM for Charts-related changes.

File: community-modules/core/src/ts/modules/moduleRegistry.ts (hunk 1 above)
Comment (taktran) [reply community-modules/core/src/ts/modules/moduleRegistry.ts]: 👍🏼 Done.

File: grid-packages/ag-grid-docs/documentation/doc-pages/grid-api/doc-interfaces.AUTO.json (hunk 2 above)
Comment (taktran) [reply grid-packages/ag-grid-docs/documentation/doc-pages/grid-api/doc-interfaces.AUTO.json:14219]: As talked about offline, we can deal with this elsewhere, as it is related to documentation generation.

File: community-modules/core/src/ts/modules/moduleRegistry.ts (hunk 1 above)
Comment (seanlandsman) [reply community-modules/core/src/ts/modules/moduleRegistry.ts]: LGTM

**Pull request merged**